File: src/main/java/x/scratch/ParentRepository.java
package x.scratch;

import org.springframework.data.jdbc.repository.query.Query;
import org.springframework.data.repository.CrudRepository;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface ParentRepository extends CrudRepository<ParentRecord, Long> {
    @Query("SELECT * FROM parent WHERE natural_id = :naturalId")
    Optional<ParentRecord> findByNaturalId(@Param("naturalId") String naturalId);
}

File: spring-boot-domain-persistence-modeling/src/main/kotlin/x/springbootdomainpersistencemodeling/ParentRepository.kt
package x.springbootdomainpersistencemodeling

import org.springframework.data.annotation.Id
import org.springframework.data.repository.CrudRepository
import java.time.Instant

interface ParentRepository : CrudRepository<ParentRecord, Long>

data class ParentRecord(
        @Id val id: Long?,
        val naturalId: String,
        var value: String,
        val version: Int,
        val createdAt: Instant,
        val updatedAt: Instant)
